auto_nag/scripts/uplift_beta.py

```diff
@@ -0,0 +1,139 @@
+
+from libmozdata import utils as lmdutils
+from libmozdata.bugzilla import Bugzilla
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+
+
+class UpliftBeta(BzCleaner):
+    def __init__(self):
+        super(UpliftBeta, self).__init__()
+        if not self.init_versions():
+            return
+
+        self.beta = self.versions["beta"]
+        self.status_central = utils.get_flag(
+            self.versions["central"], "status", "central"
+        )
+        self.status_beta = utils.get_flag(self.beta, "status", "beta")
+        self.extra_ni = {}
+
+    def description(self):
+        return "Bugs fixed in nightly but still affecting beta"
+
+    def has_assignee(self):
+        return True
+
+    def get_extra_for_needinfo_template(self):
+        return self.extra_ni
+
+    def columns(self):
+        return ["id", "summary", "assignee"]
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        assignee = bug.get("assigned_to", "")
+        if utils.is_no_assignee(assignee):
+            assignee = ""
+            nickname = ""
+        else:
+            nickname = bug["assigned_to_detail"]["nick"]
+
+        data[bugid] = {
+            "id": bugid,
+            "mail": assignee,
+            "nickname": nickname,
+            "summary": self.get_summary(bug),
+            "regressions": bug["regressions"],
+        }
+        return bug
+
+    def filter_by_regr(self, bugs):
+        # Filter the bugs which don't have any regression or where the regressions are all closed
+        def bug_handler(bug, data):
+            if bug["status"] in {"RESOLVED", "VERIFIED", "CLOSED"}:
+                data.add(bug["id"])
+
+        bugids = {r for info in bugs.values() for r in info["regressions"]}
+        if not bugids:
+            return bugs
+
+        fixed_bugs = set()
+        Bugzilla(
+            bugids=list(bugids),
+            include_fields=["id", "status"],
+            bughandler=bug_handler,
+            bugdata=fixed_bugs,
+        ).get_data().wait()
+
+        bugs_without_regr = {}
+        for bugid, info in bugs.items():
+            regs = set(info["regression"])
+            regs = regs - fixed_bugs
+            if not regs:
+                bugs_without_regr[bugid] = info
+
+        return bugs_without_regr
+
+    def get_bz_params(self, date):
+        self.date = lmdutils.get_date_ymd(date)
+        to_wait = self.get_config("days_to_wait")
+        fields = [self.status_beta, "regressions"]
+        params = {
+            "include_fields": fields,
+            "bug_type": "defect",
+            "resolution": ["---", "FIXED"],
+            "f1": self.status_central,
+            "o1": "anyexact",
+            "v1": ",".join(["fixed", "verified"]),
+            "f2": self.status_beta,
+            "o2": "anyexact",
+            "v2": "affected",
+            # Changed before 2 days ago and not changed after 2 days ago
```

Comment:
Does "last resolved" not give you what you need here?

Reply:
I'd say that `last_resolved` and this way to check are pretty similar but different.
AFAIK, `cf_last_resolved` is the last time that `resolution` is no more `---` or something like that and here we check the last change on the status flag for nightly.
Anyway, I don't care so much, finally it's up to you, so what's the best option according to your needs ?